graphqlapi/src/main/java/io/stargate/graphql/web/resources/GraphqlResourceBase.java

```diff
@@ -0,0 +1,176 @@
+/*
+ * Copyright The Stargate Authors
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ * http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+package io.stargate.graphql.web.resources;
+
+import com.fasterxml.jackson.core.JsonProcessingException;
+import com.fasterxml.jackson.databind.ObjectMapper;
+import com.google.common.base.Strings;
+import com.google.common.collect.ImmutableList;
+import com.google.common.collect.ImmutableMap;
+import graphql.ExecutionInput;
+import graphql.GraphQL;
+import io.stargate.graphql.web.HttpAwareContext;
+import io.stargate.graphql.web.models.GraphqlJsonBody;
+import java.util.Map;
+import javax.servlet.http.HttpServletRequest;
+import javax.ws.rs.Produces;
+import javax.ws.rs.QueryParam;
+import javax.ws.rs.container.AsyncResponse;
+import javax.ws.rs.container.Suspended;
+import javax.ws.rs.core.Context;
+import javax.ws.rs.core.MediaType;
+import javax.ws.rs.core.Response;
+import org.slf4j.Logger;
+import org.slf4j.LoggerFactory;
+
+/**
+ * Factors common logic to handle GraphQL queries from JAX-RS resources.
+ *
+ * @see <a href="https://graphql.org/learn/serving-over-http/">Serving (GraphQL) over HTTP</a>
+ */
+@Produces(MediaType.APPLICATION_JSON)
+public class GraphqlResourceBase {
+
+  private static final Logger LOG = LoggerFactory.getLogger(GraphqlDdlResource.class);
+  protected static final String APPLICATION_GRAPHQL = "application/graphql";
+  private static final ObjectMapper OBJECT_MAPPER = new ObjectMapper();
+
+  /**
+   * Handles a GraphQL GET request.
+   *
+   * <p>The payload is provided via URL parameters.
+   */
+  protected static void get(
+      String query,
+      String operationName,
+      String variables,
+      GraphQL graphql,
+      HttpServletRequest httpRequest,
+      AsyncResponse asyncResponse) {
+
+    if (Strings.isNullOrEmpty(query)) {
+      replyWithGraphqlError(
+          400, "You must provide a GraphQL query as a URL parameter", asyncResponse);
+    } else {
```

Comment:
If we return right after the `replyWithGraphqlError` above we don't need an else, right?

Reply:
Yes, we can remove the else's and have explicit return statements in each branch instead.

I don't really have a preference, so if that's easier to follow I can change it.

Reply:
I picked it up with golang but think it makes sense in other languages too

https://medium.com/@matryer/line-of-sight-in-code-186dd7cdea88

Reply:
Interesting post. I've simplified a few of them.